src/main/java/com/epam/grid/engine/provider/utils/slurm/healthcheck/ShowConfigCommandParser.java

```diff
@@ -0,0 +1,159 @@
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.epam.grid.engine.utils.TextConstants.SPACE;
+import static com.epam.grid.engine.utils.TextConstants.EMPTY_STRING;
+import static com.epam.grid.engine.utils.TextConstants.EQUAL_SIGN;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ShowConfigCommandParser {
+
+    private static final String UP_STATE = "UP";
+    private static final String DOWN_STATE = "DOWN";
+
+    private static final String CANT_FIND_CONNECTION = "can't find connection";
+    private static final String CANT_FIND_CONNECTION_MESSAGE = "Can`t find connection via specified port";
+    private static final String DATE_FORMAT_CHECK_TIME = "MM/dd/yyyy HH:mm:ss:";
+    private static final String DATE_FORMAT_START_TIME = "yyyy-MM-dd'T'HH:mm:ss";
+    private static final String BOOT_TIME = "BOOT_TIME";
+    private static final Long NOT_PROVIDED = 99999L;
+    private static final String DOUBLED_SPACES_REGEX = "\\s+";
+
+    public static HealthCheckInfo parseShowConfigResult(final CommandResult commandResult) {
+        validateShowConfigResponse(commandResult);
+        final List<String> stdOut = commandResult.getStdOut();
+        final StatusInfo statusInfo = parseStatusInfo(stdOut);
+        return HealthCheckInfo.builder()
+                .statusInfo(statusInfo)
+                .startTime(getStartTime(stdOut))
+                .checkTime(getCheckTime())
+                .build();
+    }
+
+    private static void validateShowConfigResponse(final CommandResult commandResult) {
+        if (CollectionUtils.isEmpty(commandResult.getStdOut())) {
+            throw new GridEngineException(HttpStatus.NOT_FOUND,
+                    String.format("Slurm error during health check. %nexitCode = %d %nstdOut: %s %nstdErr: %s",
+                            commandResult.getExitCode(), commandResult.getStdOut(), commandResult.getStdErr())
+            );
+        }
+        if (checkCantFindConnectionCase(commandResult)) {
+            throw new GridEngineException(HttpStatus.NOT_FOUND, CANT_FIND_CONNECTION_MESSAGE);
+        }
+    }
+
+    private static boolean checkCantFindConnectionCase(final CommandResult commandResult) {
+        return commandResult.getExitCode() != 0
+                && commandResult.getStdOut().get(0).contains(CANT_FIND_CONNECTION)
+                && !commandResult.getStdErr().isEmpty();
+    }
+
+    private static StatusInfo parseStatusInfo(final List<String> stdOut) {
+        final long statusCode = parseStatusCode(stdOut);
+        return StatusInfo.builder()
+                .code(statusCode)
+                .status(parseStatus(statusCode))
+                .info(parseInfo(stdOut))
+                .build();
+    }
+
+    private static long parseStatusCode(final List<String> stdOut) {
+        final long statusCode;
+        final List<String> statusString = List.of(stdOut.get(stdOut.size() - 1)
```

Comment:
it seems to me that we can simply split by `DOUBLED_SPACES_REGEX` here

Reply:
done